--- gdb-6.6/bfd/elf32-arm.c	2010-09-24 16:33:10.000000000 +0800
+++ gdb-6.6/bfd/elf32-arm.c.new	2010-09-24 16:17:07.000000000 +0800
@@ -2520,9 +2520,10 @@ find_thumb_glue (struct bfd_link_info *l
   hash = elf_link_hash_lookup
     (&(hash_table)->root, tmp_name, FALSE, FALSE, TRUE);
 
-  if (hash == NULL)
-    asprintf (error_message, _("unable to find THUMB glue '%s' for '%s'"),
-	      tmp_name, name);
+  if (hash == NULL && 
+      asprintf (error_message, _("unable to find THUMB glue '%s' for '%s'"),
+	      tmp_name, name) == -1 )
+	*error_message = (char *) bfd_errmsg (bfd_error_system_call);
 
   free (tmp_name);
 
@@ -2553,9 +2554,10 @@ find_arm_glue (struct bfd_link_info *lin
   myh = elf_link_hash_lookup
     (&(hash_table)->root, tmp_name, FALSE, FALSE, TRUE);
 
-  if (myh == NULL)
+  if (myh == NULL &&
     asprintf (error_message, _("unable to find ARM glue '%s' for '%s'"),
-	      tmp_name, name);
+	      tmp_name, name) == -1 )
+	  *error_message = (char *) bfd_errmsg (bfd_error_system_call);
 
   free (tmp_name);
 
