diff -Nrup liboil-0.3.16/configure.ac liboil-0.3.16.new/configure.ac
--- liboil-0.3.16/configure.ac	2009-03-19 13:41:04.000000000 +0800
+++ liboil-0.3.16.new/configure.ac	2010-09-24 17:22:45.000000000 +0800
@@ -26,9 +26,7 @@ AC_PROG_LIBTOOL
 AM_PROG_AS
 
 AC_CONFIG_SRCDIR([liboil/liboil.h])
-
-ACLOCAL_AMFLAGS="-I m4 $ACLOCAL_AMFLAGS"
-AC_SUBST(ACLOCAL_AMFLAGS)
+AC_CONFIG_MACRO_DIR([m4])
 
 PKG_CHECK_MODULES(GLIB, glib-2.0, HAVE_GLIB=yes, HAVE_GLIB=no)
 AC_SUBST(GLIB_LIBS)
@@ -65,9 +63,9 @@ if test "x$enable_broken_implementations
   AC_DEFINE(ENABLE_BROKEN_IMPLS, 1, [Define if compiling broken implementations])
 fi
 
-AC_ARG_ENABLE(new-abi,
-  AC_HELP_STRING([--enable-new-abi],[enable experimental 0.4 ABI]),
-  enable_new_abi=$enableval,enable_new_abi=no)
+#AC_ARG_ENABLE(new-abi,
+#  AC_HELP_STRING([--enable-new-abi],[enable experimental 0.4 ABI]),
+#  enable_new_abi=$enableval,enable_new_abi=no)
 if test "x$enable_new_abi" = xyes ; then
   AC_DEFINE(ENABLE_NEW_ABI, 1, [Define if compiling new ABI])
 fi
@@ -114,7 +112,7 @@ AC_C_BIGENDIAN
 AS_UNALIGNED_ACCESS
 
 AC_LTDL_SYMBOL_USCORE
-if test x${ac_cv_sys_symbol_underscore} = xyes ; then
+if test x${lt_cv_sys_symbol_underscore} = xyes ; then
   AC_DEFINE(HAVE_SYMBOL_UNDERSCORE, 1, [Define if C symbols have leading underscore])
 fi
 
@@ -187,7 +185,10 @@ if test x$HAVE_GCC_ASM = xyes -a x$HAVE_
     true)
 fi
 
-if test x$HAVE_GCC_ASM = xyes -a x$HAVE_ARM = xyes ; then
+AC_ARG_ENABLE(vfp,
+  AC_HELP_STRING([--enable-vfp],[compile with Vector Floating-point unit support]),
+  enable_vfp=$enableval,enable_vfp=yes)
+if test "x$enable-vfp" = xyes -a x$HAVE_GCC_ASM = xyes -a x$HAVE_ARM = xyes; then
   AS_COMPILER_FLAG(["-Wa,-mfpu=vfp"],
     [VFP_CFLAGS="$VFP_CFLAGS -mfpu=vfp"],
     true)
diff -Nrup liboil-0.3.16/liboil/arm/math_vfp_asm.S liboil-0.3.16.new/liboil/arm/math_vfp_asm.S
--- liboil-0.3.16/liboil/arm/math_vfp_asm.S	2007-11-16 10:53:47.000000000 +0800
+++ liboil-0.3.16.new/liboil/arm/math_vfp_asm.S	2010-09-24 17:22:45.000000000 +0800
@@ -24,7 +24,7 @@
  * SUCH DAMAGE.
  */
 
-#if __VFP_FP__
+#if defined(__VFP_FP__) && !defined(__SOFTFP__)
 /* 
 ** compile with -mcpu=arm1136j-s -mfpu=vfp -mfloat-abi=softfp
 **
diff -Nrup liboil-0.3.16/liboil/arm/math_vfp.c liboil-0.3.16.new/liboil/arm/math_vfp.c
--- liboil-0.3.16/liboil/arm/math_vfp.c	2008-09-26 06:36:18.000000000 +0800
+++ liboil-0.3.16.new/liboil/arm/math_vfp.c	2010-09-24 17:22:45.000000000 +0800
@@ -30,7 +30,7 @@
 #include <liboil/liboilclasses.h>
 #include <liboil/liboilfunction.h>
 
-#if __VFP_FP__
+#if defined(__VFP_FP__) && !defined(__SOFTFP__)
 
 extern void vfp_add_f32 (float *d, const float *s1, const float *s2, int n);
 extern void vfp_add_f64 (double *d, const double *s1, const double *s2, int n);
diff -Nrup liboil-0.3.16/liboil/Makefile.am liboil-0.3.16.new/liboil/Makefile.am
--- liboil-0.3.16/liboil/Makefile.am	2009-02-27 03:35:39.000000000 +0800
+++ liboil-0.3.16.new/liboil/Makefile.am	2010-09-24 17:22:45.000000000 +0800
@@ -39,6 +39,10 @@ if HAVE_POWERPC
 if HAVE_GCC_ASM
 SUBDIRS += powerpc motovec
 libs += powerpc/libpowerpc.la motovec/libmotovec.la
+
+# motovec doesn't compile on snow leopard, and isn't useful elsewhere
+#SUBDIRS += motovec
+#libs += motovec/libmotovec.la
 if HAVE_ASM_BLOCKS
 SUBDIRS += powerpc_asm_blocks
 libs += powerpc/libpowerpc_asm_blocks.la
diff -Nrup liboil-0.3.16/liboil/sse/math_sse.c liboil-0.3.16.new/liboil/sse/math_sse.c
--- liboil-0.3.16/liboil/sse/math_sse.c	2009-03-09 02:00:10.000000000 +0800
+++ liboil-0.3.16.new/liboil/sse/math_sse.c	2010-09-24 17:22:45.000000000 +0800
@@ -353,7 +353,7 @@ scalarmultiply_f64_ns_sse2 (double *dest
   for (; ((long)dest & 15) && (n > 0); n--) {
     *dest++ = *src1++ * *val;
   }
-  xmm1 = _mm_load_pd1(val);
+  xmm1 = _mm_load1_pd(val);
   for (; n >= 2; n -= 2) {
     __m128d xmm0;
     xmm0 = _mm_loadu_pd(src1);
diff -Nrup liboil-0.3.16/liboil/sse/math_sse_unroll2.c liboil-0.3.16.new/liboil/sse/math_sse_unroll2.c
--- liboil-0.3.16/liboil/sse/math_sse_unroll2.c	2009-03-09 02:00:16.000000000 +0800
+++ liboil-0.3.16.new/liboil/sse/math_sse_unroll2.c	2010-09-24 17:22:45.000000000 +0800
@@ -326,7 +326,7 @@ scalarmultiply_f64_ns_sse2_unroll2 (doub
   for (; ((long)dest & 15) && (n > 0); n--) {
     *dest++ = *src1++ * *val;
   }
-  xmm1 = _mm_load_pd1(val);
+  xmm1 = _mm_load1_pd(val);
   for (; n >= 4; n -= 4) {
     __m128d xmm0;
     xmm0 = _mm_loadu_pd(src1);
diff -Nrup liboil-0.3.16/Makefile.am liboil-0.3.16.new/Makefile.am
--- liboil-0.3.16/Makefile.am	2008-09-26 03:29:54.000000000 +0800
+++ liboil-0.3.16.new/Makefile.am	2010-09-24 17:22:45.000000000 +0800
@@ -19,5 +19,5 @@ BUILT_SOURCES=liboil-$(LIBOIL_MAJORMINOR
 
 CLEANFILES = liboil-$(LIBOIL_MAJORMINOR).pc liboil-$(LIBOIL_MAJORMINOR)-uninstalled.pc
 
-ACLOCAL_FLAGS = -I m4
+ACLOCAL_AMFLAGS = -I m4
 
