--- tslib-1.0/tests/ts_calibrate.c	2010-09-19 18:57:18.000000000 +0800
+++ tslib-1.0.new/tests/ts_calibrate.c	2010-09-19 18:55:49.000000000 +0800
@@ -24,6 +24,8 @@
 #include <linux/kd.h>
 #include <linux/vt.h>
 #include <linux/fb.h>
+#include <sys/types.h>
+#include <sys/stat.h>
 
 #include "tslib.h"
 
@@ -224,9 +226,9 @@
 		for (i = 0; i < 7; i++) printf("%d ", cal.a [i]);
 		printf("\n");
 		if ((calfile = getenv("TSLIB_CALIBFILE")) != NULL) {
-			cal_fd = open (calfile, O_CREAT | O_RDWR);
+			cal_fd = open (calfile, O_CREAT | O_RDWR, 0x0666);
 		} else {
-			cal_fd = open ("/etc/pointercal", O_CREAT | O_RDWR);
+			cal_fd = open ("/etc/pointercal", O_CREAT | O_RDWR, 0x0666);
 		}
 		sprintf (cal_buffer,"%d %d %d %d %d %d %d",
 			 cal.a[1], cal.a[2], cal.a[0],
