diff --exclude CVS --exclude .git -uNr udev-117/etc/udev/ltib/10-imx.rules udev-117.modified/etc/udev/ltib/10-imx.rules
--- udev-117/etc/udev/ltib/10-imx.rules	2010-03-17 11:44:26.252943798 +0800
+++ udev-117.modified/etc/udev/ltib/10-imx.rules	2010-03-17 11:43:41.941103600 +0800
@@ -5,6 +5,7 @@
 SUBSYSTEM=="input" KERNEL=="event*" SYSFS{name}=="mpr084",     SYMLINK+="input/keyboard0"
 SUBSYSTEM=="input" KERNEL=="event*" SYSFS{name}=="tsc2007",    SYMLINK+="input/ts0"
 SUBSYSTEM=="input" KERNEL=="event*" SYSFS{name} =="STMP3XXX touchscreen",    SYMLINK+="input/ts0"
+SUBSYSTEM=="input" KERNEL=="event*" SYSFS{name} =="MXS touchscreen",    SYMLINK+="input/ts0"
 
 # The long class name gets cut off to be mc13783_connectiv in 
 # /sys/class/mc13783_connectivity/mc13783_connectivit
